### ISSUE TITLE
Add new loss functions on training

### ISSUE THREAD

**Issue by sagnik1511** [sagnik1511/Tabular-AutoML#9]
1. Add 3 loss functions for both regression and classification problem types.
2.  Add them similarly to how the model scores are stored. See [here](https://github.com/sagnik1511/Tabular-AutoML/blob/46e385dcc5fe8a67b4f87861cb7ad3b84d8420d6/tab_automl/automl/training.py#L67)
2. Add proper comments.
3. If new functions are needed for the loss functions, store them on [`tab_automl.utils.training`](https://github.com/sagnik1511/Tabular-AutoML/blob/main/tab_automl/utils/training.py) .
4. Update the requirements if new libraries are being used.

Follow contributing guidelines on [README.md](https://github.com/sagnik1511/Tabular-AutoML/blob/main/README.md)

**Comment by das-ankur** [comment]
I want to work on this issue under JWOC

**Comment by sagnik1511** [comment]
Please make sure that you add atleast 3 loss functions. 
Good Luck!

**Comment by das-ankur** [comment]
Hello, I need help. Can you please tell me, do I need to store the py files in utils?

**Comment by sagnik1511** [comment]
1. First store the functions inside `utils.training` 
2. Use those functions on the main training file. 
3. If you want another function to iterate through those loss functions, create another function under the class Trainer.

**Comment by sagnik1511** [comment]
Hey @das-ankur, any update on the task?

**Comment by sagnik1511** [comment]
As 5 days have been completed, @das-ankur please update the task details so that I can merge that, no response will be recorded as an incomplete task.

**Comment by das-ankur** [comment]
I'm really sorry for the late reply, I have had a fever for the last two days. I will do PR by Tuesday.

**Comment by sagnik1511** [comment]
Unassigned @das-ankur
